repose-aggregator/core/repose-core/src/test/java/org/openrepose/core/services/reporting/destinations/DestinationInfoLogicTest.java

```diff
@@ -117,16 +116,36 @@ public void shouldReturnZeroIfNoResponseTime() {
 
         @Test
         public void shouldGetThroughput() throws InterruptedException {
-            long totalResponses = 1000l;
-
-            for (int i = 0; i < 1000; i++) {
+            long totalResponses = 1000L;
+
+            // As noted below, this value helps normalize the test result by minimizing the effect of test runtime.
+            // This value should be significantly longer than the runtime of the test itself.
+            // Since this test is currently running in no more than 2ms, 1s should be sufficient.
+            // todo: rather than trying to account for test runtime, use a time service that can be injected for
+            // todo: more exact testing
+            long sleepTime = 1000L;
+
+            // These values are the maximum allowable variance in test runtime.
+            // Since this test should definitely complete within 10ms, we use that.
+            // We then convert the variance to seconds and adjust for the sleep time to calculate
+            // the acceptable throughput variance.
+            double maxVarianceInMs = 10.0;
+            double maxThroughputVariance = maxVarianceInMs / 1000.0 * sleepTime;
+
+            for (int i = 0; i < totalResponses; i++) {
                 destinationInfoLogic.incrementResponseCount();
             }
 
+            // By sleeping for a certain amount of time that is significantly longer than the expected runtime of this
+            // test, we can minimize the potential difference between the expected and actual throughput.
+            // That is, as wait time increases while runtime is relatively constant (or, at least, with minimal variance),
+            // our expected and actual throughput converge.
+            Thread.sleep(sleepTime);
```

Comment:
This solution actually solves two problems:

1. Precision. By waiting, we don't have to worry about calculating `Infinity` due to `System.currentTimeMillis()` returning the same time (i.e., the test running within a single millisecond).
1. Variance. By waiting, we decrease the effect of test runtime on our calculations.

That being said, I hate all of this. If I had the time and inclination, I would either remove this functionality (if not necessary/desired), or create a time service that could be injected for more exact testing here. Or maybe something else that I haven't thought of.